### PR TITLE
Added loging level through mqtt

### PR DESCRIPTION
**Pull request by ciotlosm**
Added loging level through mqtt

Just logging entity (not libraries) and no documentation added in this PR. Couldn't figure out how to print nicely log level as didn't want to have it as "critical" the information that you're switching log level. 

Implements https://github.com/Koenkk/zigbee2mqtt/issues/81 parially

- [x] Allow setting log level through mqtt
- [x] Allow setting log level through configuration
- [x] Allow setting log level for dependencies (where possible, like https://github.com/Koenkk/zigbee2mqtt/issues/81#issuecomment-394113688)
- [x] Force debug on start-up for `logger` when received `DEBUG` env variable
- [x] Documentation on both mqtt and configuration for log_level

### CODE REVIEW

File: lib/controller.js [hunk 1]
```diff
@@ -322,6 +324,15 @@ class Controller {
 
         if (option === 'permit_join') {
             this.zigbee.permitJoin(message.toString().toLowerCase() === 'true');
+        } else if (option === 'log_level') {
+            const level = message.toString().toLowerCase();
+            if (allowedLogLevels.includes(level)) {
+                logger.warn(`Switching log level to '${level}'`);
```
Comment (Koenkk) [inline lib/controller.js]: This is not a warning, I think `logger.info` fits better.

File: lib/controller.js [hunk 2]
```diff
@@ -322,6 +324,15 @@ class Controller {
...
+                logger.transports.console.level = level;
+                logger.transports.file.level = level;
+            } else {
+                logger.error(`Could not set log level to '${level}'`);
```
Comment (Koenkk) [inline lib/controller.js]: Would be nice to log the possible values. e.g.
```
logger.error(`Could not set log level to '${level}', possible levels are ${allowedLogLevels}`);
```

Comment (ciotlosm) [reply lib/controller.js]: Added

File: lib/controller.js (hunk 1 above)
Comment (ciotlosm) [reply lib/controller.js]: If you set this to info, but you set log level to warn you miss the message that log level has changed. 
If it is acceptable happy to change it.

**Comment by ciotlosm** [comment]
It seems that even if https://github.com/visionmedia/debug is capable of dynamic [enable/disable](https://github.com/visionmedia/debug#set-dynamically), it's not possible to do it across all because dependencies have debug pinned to old version without support :(

I will try a few more workarounds but I doubt I will be in luck

**Comment by Koenkk** [comment]
Is this needed for Allow setting log level for dependencies (where possible, like #81 (comment))? I think this is a 'nice to have'.

**Comment by ciotlosm** [comment]
I'll move debug messages in zigbee2mqtt to logger and let the rest use visionmedia/debug. Any reason for using debug [here](https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/controller.js#L200)?

**Comment by ciotlosm** [comment]
Will look to move debug messages for dependencies into a separate PR if I find a workaround.

File: lib/controller.js (hunk 1 above)
Comment (ciotlosm) [reply lib/controller.js]: fixed

**Comment by ciotlosm** [comment]
@Koenkk Updated also documentation. Can you please have a look? 
The only use case I haven't covered yet it's to set DEBUG to * when receiving 'debug' from configuration.yaml. Not sure if it's possible, but will give it a try.

**Comment by Koenkk** [comment]
I've updated: https://github.com/Koenkk/zigbee2mqtt/wiki/MQTT-topics-and-message-structure#zigbee2mqttbridgeconfiglog_level

**Comment by Koenkk** [comment]
Thanks!

**Comment by ciotlosm** [comment]
Perfect. Should I close the logging issue?

**Pull request merged**